File: assignment2/part1/section2/scopes_closures.py
def make_counter(start):
    count = start

    def increment(step):
        nonlocal count
        count += step
        return count

    return increment


counter = make_counter(10)
print(counter(1))
print(counter(5))

items = [1, 2, 3]
closures = []
for item in items:
    closures.append(lambda: item)

print([fn() for fn in closures])

File: assignment2/part1/section2/scopes_closures.js
function makeCounter(start) {
    let count = start;

    return function increment(step) {
        count += step;
        return count;
    };
}

const counter = makeCounter(10);
console.log(counter(1));
console.log(counter(5));

const closures = [];
for (let item of [1, 2, 3]) {
    closures.push(() => item);
}

console.log(closures.map((fn) => fn()));
